### PR TITLE
handle DCACCOUNT and DCLOGIN schemes on new intents

### LINKED ISSUE #2365: does not register for `DCACCOUNT:` uri scheme in system
this should work to make the qr code on webxdc.org clickable

also see:
https://github.com/deltachat/deltachat-ios/pull/1640
https://github.com/deltachat/deltachat-desktop/pull/2823

### PR DESCRIPTION
**Pull request by adbenitez**
close #2365

### CODE REVIEW

File: src/org/thoughtcrime/securesms/WelcomeActivity.java [hunk 1]
```diff
@@ -96,6 +101,32 @@ private void registerForEvents() {
...
     }
 
+    private void handleIntent() {
+        if (getIntent() != null && Intent.ACTION_VIEW.equals(getIntent().getAction())) {
+            Uri uri = getIntent().getData();
+            if (uri == null) return;
+
+            if (uri.getScheme().equalsIgnoreCase(DCACCOUNT)) {
+                String uriString = uri.toString();
+                uriString = uriString.replaceFirst(DCACCOUNT, DCACCOUNT.toUpperCase());
```
Comment (Hocuri) [inline src/org/thoughtcrime/securesms/WelcomeActivity.java:111]: Is this line really necessary? Also, it seems to be the only difference to the `else if` block?

**Comment by Hocuri** [comment]
Nice! Will have to wait for https://github.com/deltachat/deltachat-core-rust/pull/3541 to be merged first, I guess?

**Comment by Hocuri** [comment]
Did you test if this works when there already is a configured account?

File: src/org/thoughtcrime/securesms/WelcomeActivity.java (hunk 1 above)
Comment (adbenitez) [reply src/org/thoughtcrime/securesms/WelcomeActivity.java]: simplified :heavy_check_mark:

**Comment by adbenitez** [comment]
> Did you test if this works when there already is a configured account?

yes

**Review by Hocuri — APPROVED** [review]

**Pull request merged**